mewarpx/mewarpx/mwxrun.py

```diff
@@ -55,10 +55,6 @@ def __init__(self):
         self.simulation = picmi.Simulation(verbose=0)
         # make a shorthand for simulation.extension since we use it a lot
         self.sim_ext = self.simulation.extension
-        # this will be the first after init function to be called
-        callbacks.installafterinit(self._after_init)
-        # grab the simulation geometry from simulation.extension
-        self._set_geom_props()
 
     def init_grid(self, lower_bound, upper_bound, number_of_cells, **kwargs):
```

Comment:
Are there easy ways to factor out 1 or 2 subfunctions from this? We're at > 100 lines, which is not terrible, but it's becoming a long & multi-faceted function...
I don't mean changing API, I just mean taking out chunks of code into separate functions.